### PR TITLE
SignUp and Welcome Screen

### PR DESCRIPTION
**Pull request by sivaShilpa**
### Done: 

1. Created the Sign Up Page
2. Created the reusable Input Field Component
3. Created the reusable third party button component. 
4. Used the button component as the sign up button
5. Fixed the back arrow and the name on the header
6. Fixed the test files. 
7. Fixed the login Link. Now it navigates to login screen

### Notes: 

- Please watch this Jam video https://jam.dev/c/766bd9ee-73f5-41b2-ac67-f64b5c6d1d68

### CODE REVIEW

**Review by lindentree — COMMENTED** [review]
@MelisaIm This PR should probably just supersede the other PR and be approved first.

@sivaShilpa In the future, PRs should be smaller and more atomic(self-contained). Packing in all changes from other branches makes reviewing too difficult. In a company environment, you would probably be asked to break this PR down back into the Login Page PR and a Signup Page PR.

File: src/components/InputField.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { useFonts } from 'expo-font';
+import { Text, StyleSheet, Pressable, TextInput, View } from 'react-native';
+
+type Props = {
+  label: string;
+  isPassword: boolean;
+  onChange: (text: string) => void;
+};
+
+const InputField: React.FC<Props> = ({ label, isPassword, onChange }: Props) => {
```
Comment (lindentree) [inline src/components/InputField.tsx:11]: I strongly disagree with the idea of creating an InputField component abstracted away from the form that contains it. This should be its own PR at the very least. It also makes it harder to implement input validation(eg passwords of certain length/characters, or valid email addresses/valid names etc)

File: src/screens/SignUpScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
+import { Text, StyleSheet, View, TouchableOpacity } from 'react-native';
+import React from 'react';
+import InputField from '../components/InputField';
+import { useFonts } from 'expo-font';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import ButtonComponent from '../components/ButtonComponent';
+import ThirdPartyButton from '../components/ThirdPartyButton';
+
+const SignUpScreen = ({ navigation }: { navigation: any }) => {
+  const [fullName, setFullName] = React.useState('');
```
Comment (lindentree) [inline src/screens/SignUpScreen.tsx:10]: probably better to use a single object for the state

File: src/components/InputField.tsx (hunk 1 above)
Comment (sivaShilpa) [reply src/components/InputField.tsx:11]: I had the button component added to the Welcome Screen PR, and it was accepted. So thought this was okay. Will keep it more organized in the future in separate PR's. Please accept the Input Field along with the third party button component in this PR this time, as it is hard and confusing to separate them to different PR's now. Will keep PR's small and separate in the future.

Comment (lindentree) [reply src/components/InputField.tsx:11]: That will be up to @MelisaIm after she reviews this code. This code is not adequately tested, render tests are just for example. This is also true for the other components. I personally am reluctant to add so many untested components all at once.

Comment (sivaShilpa) [reply src/components/InputField.tsx:11]: Writing testing files is something I am new to and would love to spend some time learning and working on them.

**Comment by MelisaIm** [comment]
Issue #4

**Comment by sivaShilpa** [comment]
Redoing the entire Welcome Screen and Sign Up Screen to get them organized in separate PR's. As a result I am closing this PR for now.
